EZSwiftExtensionsTests/EnumTests.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  EnumTests.swift
+//  EZSwiftExtensions
+//
+//  Created by furuyan on 2017/08/26.
+//  Copyright © 2017年 Goktug Yilmaz. All rights reserved.
+//
+
+import XCTest
+@testable import EZSwiftExtensions
+
+class EnumTests: XCTestCase {
+    enum Fruit: EnumCollection {
+        case apple, orange, banana
+    }
+
+    func testEnumerated() {
+        for (i, fruit) in Fruit.enumrated() {
+            switch i {
+            case 0:
```

Comment:
How is this more useful for than pattern matching on enums in swift ? : https://appventure.me/2015/08/20/swift-pattern-matching-in-detail/#sec-3-5? The indices hide the order without a lookup. The pattern matching syntax allows a proper destruct on the values and is vastly more readable. 

I do not think I like this extension, since there is already a cleaner way to match on enums in the language.